code/modules/mob/living/living_emote.dm

```diff
@@ -0,0 +1,418 @@
+
+/datum/emote/living/sulk
+	key = "sulk"
+	key_third_person = "sulks"
+	message = "sulks down sadly."
+
+/datum/emote/living/sway
+	key = "sway"
+	key_third_person = "sways"
+	message = "sways around dizzily."
+
+/datum/emote/living/swear
+	key = "swear"
+	key_third_person = "swears"
+	message = "says a swear word!"
+	message_param = "says a swear word at %t!"
+	message_mime = "makes a rude gesture!"
+	message_simple = "makes an angry noise!"
+	message_robot = "makes a particularly offensive series of beeps!"
+	message_postfix = "at %t!"
+	emote_type = EMOTE_AUDIBLE | EMOTE_MOUTH
+
+/datum/emote/living/tilt
+	key = "tilt"
+	key_third_person = "tilts"
+	message = "tilts their head to the side."
+
+/datum/emote/living/tremble
+	key = "tremble"
+	key_third_person = "trembles"
+	message = "trembles in fear!"
+
+/datum/emote/living/twitch
+	key = "twitch"
+	key_third_person = "twitches"
+	message = "twitches violently."
+
+/datum/emote/living/twitch_s
+	key = "twitch_s"
+	message = "twitches."
+
+
+/datum/emote/living/whimper
+	key = "whimper"
+	key_third_person = "whimpers"
+	message = "whimpers."
+	message_mime = "appears hurt."
+	emote_type = EMOTE_AUDIBLE | EMOTE_MOUTH
+	muzzled_noises = list("weak", "pathetic")
+
+/datum/emote/living/wsmile
+	key = "wsmile"
+	key_third_person = "wsmiles"
+	message = "smiles weakly."
+
+/datum/emote/living/custom
+	key = "me"
+	key_third_person = "custom"
+	message = null
+	mob_type_blacklist_typecache = list(
+		/mob/living/carbon/brain,	// nice try
+		/mob/living/captive_brain
+	)
+
+/datum/emote/living/custom/can_run_emote(mob/user, status_check, intentional)
+	. = ..()
+	return . && intentional
+
+/datum/emote/living/custom/proc/check_invalid(mob/user, input)
+	var/static/regex/stop_bad_mime = regex(@"says|exclaims|yells|asks")
+	if(stop_bad_mime.Find(input, 1, 1))
+		to_chat(user, "<span class='danger'>Invalid emote.</span>")
+		return TRUE
+	return FALSE
+
+/datum/emote/living/custom/run_emote(mob/user, params, type_override = null, intentional = FALSE)
+	var/custom_emote
+	var/custom_emote_type
+
+	if(QDELETED(user))
+		return FALSE
+	else if(check_mute(user?.client?.ckey, MUTE_IC))
+		to_chat(user, "<span class='boldwarning'>You cannot send IC messages (muted).</span>")
+		return FALSE
+	else if(!params)
+		custom_emote = copytext(sanitize(input("Choose an emote to display.") as text|null), 1, MAX_MESSAGE_LEN)
+		if(custom_emote && !check_invalid(user, custom_emote))
+			var/type = input("Is this a visible or hearable emote?") as null|anything in list("Visible", "Hearable")
+			switch(type)
+				if("Visible")
+					custom_emote_type = EMOTE_VISIBLE
+				if("Hearable")
+					custom_emote_type = EMOTE_AUDIBLE
+				else
+					to_chat(user,"<span class='warning'>Unable to use this emote, must be either hearable or visible.</span>")
+					return
+	else
+		custom_emote = params
+		if(type_override)
+			custom_emote_type = type_override
+	message = custom_emote
+	emote_type = custom_emote_type
+	. = ..()
```

Comment:
Any reason you have the parent call this low?

Reply:
Yeah, the reason is so that our changes to the message get passed through the message sending logic in the parent call.